sdk/core/core-http-compat/review/core-http-compat.api.md

```diff
@@ -0,0 +1,44 @@
+## API Report File for "@azure/core-http-compat"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { CommonClientOptions } from '@azure/core-client';
+import { ServiceClient } from '@azure/core-client';
+import { ServiceClientOptions } from '@azure/core-client';
+
+// @public (undocumented)
+export interface KeepAliveOptions {
+    enable?: boolean;
+}
+
+// @public (undocumented)
+export interface RedirectPolicyOptions {
+    // (undocumented)
+    handleRedirects?: boolean;
+    maxRetries?: number;
+}
+
+// @public (undocumented)
+export class ShimClient extends ServiceClient {
+    constructor(options: ShimClientOptions);
+}
+
+// @public (undocumented)
+export interface ShimClientOptions extends ServiceClientOptions {
```

Comment:
Since we end up needing both this and `ShimCommonClientOptions` I wonder if we should do something like

```ts
export interface ShimOptions {
   keepAliveOptions?: KeepAliveOptions;
    redirectOptions?: RedirectPolicyOptions;
}

export type ShimClientOptions = ServiceClientOptions & ShimOptions;

export type ShimCommonClientOptions = CommonClientOptions & ShimOptions;
```

So they'll always stay in sync. Or is that too much type math? /cc @witemple-msft  for his opinion

Reply:
Done

Reply:
Done